agent/engine/dependencygraph/graph.go

```diff
@@ -17,7 +17,17 @@ import (
 	"strings"
 
 	"github.com/aws/amazon-ecs-agent/agent/api"
+	"github.com/aws/amazon-ecs-agent/agent/credentials"
 	log "github.com/cihub/seelog"
+	"github.com/pkg/errors"
+)
+
+var (
+	ExecutionCredentialsNotResolved = errors.New("dependency graph: container execution credentials not available")
```

Comment:
A bunch of these are not referenced outside the package I think. Can you make those to be package scoped instead? Also, please add documentation for all public error codes.